op-supervisor/supervisor/backend/processors/chain_processor.go

```diff
@@ -151,34 +161,126 @@ func (s *ChainProcessor) work() {
 	}
 }
 
-func (s *ChainProcessor) update(nextNum uint64) error {
+// adjustParallelism elasticly adjusts the number of fetcher threads to be used based on the number of blocks processed
+func (s *ChainProcessor) adjustParallelism(processed int) {
```

Comment:
I'm not sure if I understand the purpose of the dynamic number of fetching threads, and why it's updating it based on the previous number of results. Shouldn't we just aim to process between the starting point and the expected head block number, with some maximum parallelism?

A simple `min(s.lastHead.Number - s.nextNum(), maxThreads)` would be sufficient no? And abort the range-update if there's nothing to fetch, instead of the lower fetcher thread bound.

Reply:
I've removed the elastic management.